controllers/flowlogspipeline/flp_service_monitor.go

```diff
@@ -0,0 +1,48 @@
+package flowlogspipeline
+
+import (
+	"context"
+
+	"github.com/netobserv/network-observability-operator/controllers/constants"
+	"github.com/netobserv/network-observability-operator/controllers/reconcilers"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func AddPrometheusServiceMonitor(ctx context.Context, b *builder, cl reconcilers.ClientHelper) error {
+	serviceMonitorObject := buildPrometheusServiceMonitorObject(b)
+	// apply object to kubernetes
+	if err := cl.CreateOwned(ctx, serviceMonitorObject); err != nil {
+		return err
+	}
+	return nil
+}
+
+func buildPrometheusServiceMonitorObject(b *builder) *monitoringv1.ServiceMonitor {
+	flpServiceMonitorObject := monitoringv1.ServiceMonitor{
```

Comment:
since this rely on `ServiceMonitor` custom resource, we should either ensure `servicemonitors.monitoring.coreos.com` CustomResourceDefinition exists or let the user configure this from our CRD.

Else it will crash as you can see in the [CI running on kind](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/netobserv_network-observability-operator/188/pull-ci-netobserv-network-observability-operator-main-ci/1586679116831133696#1:build-log.txt%3A500):
`"error": "no kind is registered for the type v1.ServiceMonitor in scheme \"k8s.io/client-go/kubernetes/scheme/register.go:74\"`

Reply:
added check for existence of `servicemonitors.monitoring.coreos.com`.

Reply:
thanks @KalmanMeth !